### ISSUE TITLE
Error upgrading database using migrate-from-acl: NotNullConstraintViolationException

### ISSUE THREAD

**Issue by dani-li** [mapbender/mapbender#1597]
Describe the bug
During the process of upgrading Mapbender, I encountered a critical error when executing the command php bin/console mapbender:security:migrate-from-acl. The error message indicates a NotNullConstraintViolationException related to a NULL value in the subject_domain column of the fom_permission table.

To Reproduce

Upgrade the database by executing php bin/console mapbender:database:upgrade.
Execute the security migration command php bin/console mapbender:security:migrate-from-acl.
Observe the error message indicating a NULL value violation in the subject_domain column.
Expected behavior
The security migration should complete successfully without any NULL value violations in the subject_domain column.

Versions:
Mapbender: 4.0.0-RC1
Database: PostgreSQL 15.4
Apache: 2.4.59 (Win64) mod_fcgid/2.3.10-dev
PHP Version: 8.3.7

Error Message:
11:56:26 CRITICAL  [console] Error thrown while running command "mapbender:security:migrate-from-acl". Message: "An exception occurred while executing a query: SQLSTATE[23502]: Not null violation: 7 FEHLER:  NULL-Wert in Spalte »subject_domain« von Relation »fom_permission« verletzt Not-Null-Constraint\nDETAIL:  Fehlgeschlagene Zeile enthält (28, null, null, null, 32, null, null, application, null, view)." ["exception" => Doctrine\DBAL\Exception\NotNullConstraintViolationException^ { …},"command" => "mapbender:security:migrate-from-acl","message" => """  An exception occurred while executing a query: SQLSTATE[23502]: Not null violation: 7 FEHLER:  NULL-Wert in Spalte »subject_domain« von Relation »fom_permission« verletzt Not-Null-Constraint\n  DETAIL:  Fehlgeschlagene Zeile enthält (28, null, null, null, 32, null, null, application, null, view).  """]
{"message":"Command \"mapbender:security:migrate-from-acl\" exited with code \"7\"","context":{"command":"mapbender:security:migrate-from-acl","code":7},"level":100,"level_name":"DEBUG","channel":"console","datetime":"2024-06-07T11:56:26.112079+02:00","extra":{}}

**Comment by astroidex** [comment]
hello @dani-li ,
thank you for your feedback. I will check your information and reach out for more information if needed.

**Comment by Phocacius** [comment]
Hello @dani-li, would it be possible to send me a dump of your acl_* tables from before the update? That would help in investigating where exactly something went wrong. You can send it to thorsten.hack@wheregroup.com if you don't want to upload it publicly. Thank you!

**Comment by Phocacius** [comment]
Thanks @dani-li for sending over the files.

The issue occurred when a group's name or a username has been updated after assigning a permission to it. The old ACL system had the names saved instead of the ids and did not have an updating system. With the new permission system that won't happen, since the ids of users and groups are referenced and entries will automatically be deleted when a group or user is deleted.

The issue can't be easily fixed since there's no way to tell which group has been renamed. As a workaround, a warning is shown saying which group or user can't be found. Manual investigation and recreating the permission entry/entries affected is necessary then unfortunately. The command will now run without throwing an error though.

**Comment by dani-li** [comment]
Hello Thorsten,

After encountering the error:
**"WARNING: could not find group [group ID], defined in entry id [entry ID],**"
I deleted the entry with ID [entry ID] directly from the acl_entries table in the database, and it resolved the issue. This approach was more effective because after running the command bin/console mapbender:database:upgrade, I was unable to open Mapbender anymore (The server returned a "500 Internal Server Error.") even after clearing the cache.

Thank you very much for your help. I recommend describing this error in the documentation.

Additionally, I noticed that the official documentation does not include the commands
**bin/console mapbender:database:upgrade**
and
**bin/console mapbender:security:migrate-from-acl**
before
**bin/console doctrine:schema:update --complete --force**
as part of the process, unlike the Upgrading Guide on GitHub. Even though the documentation recommends reading the GitHub page, I think it would be beneficial to include these steps directly in the official documentation to avoid errors for other users, as all users upgrading from a version under 4.0 should run these commands.

Best regards,
Danilo de Lima
